spec/features/users_spec.rb

```diff
@@ -0,0 +1,48 @@
+require "rails_helper"
+
+feature "can sign up for an account" do
+  scenario "with valid attributes" do
+    visit new_user_path
+
+    fill_in "user_name", with: "Mary"
+    fill_in "user_email", with: "Mary@example.com"
+    fill_in "user_password", with: "secret"
+    click_on I18n.t("users.new.submit")
+
+    expect(page).to have_text(I18n.t("users.show.welcome", name: "Mary"))
+  end
+end
+
+feature "user can see their show page" do
+  scenario "signing up with valid attributes" do
+    user = create(:user, name: "John")
+
+    visit user_path(user)
+
+    expect(page).to have_text(I18n.t("users.show.welcome", name: user.name))
+  end
+end
+
+feature "user can update their profile" do
+  scenario "logging in with valid attributes" do
+    user = create(:user)
+
+    visit edit_user_path(user)
+
+    fill_in "user_email", with: "JT@example.com"
+    click_on I18n.t("users.edit.submit")
+
+    expect(page).to have_text("JT@example.com")
+  end
+end
+
+feature "user can delete their account" do
+  scenario "with valid attributes" do
+    user = create(:user)
+
+    visit edit_user_path(user)
+    click_on I18n.t("users.edit.delete")
+
+    expect(page).to have_text(I18n.t("home.show.welcome"))
```

Comment:
I would check that there is a flash message here on the page that a record has been destroyed.

Reply:
Thanks Elle, I have added the test but I couldn't find a way to use responders in my controller destroy method, I fell back on flash message and redirect_to. Any clues on this would be most welcome.